javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -0,0 +1,1729 @@
+          (multiplyByTenPow(scaledDividend0, -diffScale.toInt), scaledDivisor0)
+        else
+          (scaledDividend0,scaledDivisor0)
+
+      divideBigIntegers(scaledDividend, scaledDivisor, scale, roundingMode)
+    }
+
+    if (this.BitLength < 64 && divisor.BitLength < 64) {
+      val lpt = LongTenPows(diffScale.toInt)
+      val lptLen = LongTenPows.length
+      val lptbLen = LongTenPowsBitLength(diffScale.toInt)
+      val lptMinus = LongTenPows(-diffScale.toInt)
+      val lptbMinusLen = LongTenPowsBitLength(-diffScale.toInt)
+
+      if (diffScale == 0) {
+        val div = divisor.SmallValue
+        dividePrimitiveLongs(SmallValue, div, scale, roundingMode)
+      } else if (diffScale > 0) {
+        if (diffScale < lptLen && divisor.BitLength + lptbLen < 64) {
+          val div = divisor.SmallValue * lpt
+          dividePrimitiveLongs(SmallValue, div, scale, roundingMode)
+        } else {
+          default()
+        }
+      } else if (-diffScale < lptLen && this.BitLength + lptbMinusLen < 64) {
+        val div = SmallValue * lptMinus
+        dividePrimitiveLongs(div, divisor.SmallValue, scale, roundingMode)
+      } else {
+        default()
+      }
+    } else {
+      default()
+    }
+  }
+
+  def divide(divisor: BigDecimal, roundingMode: Int): BigDecimal =
+    divide(divisor, Scale, RoundingMode.valueOf(roundingMode))
+
+  def divide(divisor: BigDecimal, roundingMode: RoundingMode): BigDecimal =
+    divide(divisor, Scale, roundingMode)
+
+  def divide(divisor: BigDecimal): BigDecimal = {
+    val thisUnscaled = this.getUnscaledValue
+    val diffScale:Long = Scale.toLong - divisor.Scale
+
+    if (divisor.isZero) {
+      throw new ArithmeticException("Division by zero")
+    } else if (thisUnscaled.signum() == 0) {
+      zeroScaledBy(diffScale)
+    } else {
+      val divisorUnscaled = divisor.getUnscaledValue
+      val lastPow = BigFivePows.length - 1
+      val gcd = thisUnscaled.gcd(divisorUnscaled) // To divide both by the GCD
+      val p = thisUnscaled.divide(gcd)
+      val q1 = divisorUnscaled.divide(gcd)
+      // To simplify all "2" factors of q, dividing by 2^k
+      val k = q1.getLowestSetBit // number of factors "2" in 'q'
+
+      @inline
+      @tailrec
+      def loop(i: Int, q: BigInteger, l: Int ): (BigInteger, Int) = {
+        val quotAndRem = q.divideAndRemainder(BigFivePows(i))
+        if (quotAndRem(1).signum() == 0)
+          loop(if (i < lastPow) i + 1 else i,  quotAndRem(0), l + i)
+        else if (i != 1)
+          loop(1, q, l)
+        else
+          (q, l)
+      }
+
+      //q simplifies all "5" factors of q1, dividing by 5^l
+      //l number of factors "5" in divisorUnscaled
+      val (q, l) = loop(1, q1.shiftRight(k), 0)
+
+      // If  abs(q) != 1  then the quotient is periodic
+      if (q.abs() != BigInteger.ONE) {
+        val msg =
+          "Non-terminating decimal expansion; no exact representable decimal result"
+        throw new ArithmeticException(msg)
```

Comment:
Don't use an intermediate `msg`. Simply put the message in the (), on the next line with 4 spaces:
```
throw new ArithmeticException(
    "Non-term ...")
```